aptos-move/framework/supra-framework/sources/automation_registry.move

```diff
@@ -0,0 +1,148 @@
+    use supra_framework::event;
+    use supra_framework::transaction_context;
+    use supra_framework::system_addresses;
+
+    /// Registry Id not found
+    const EREGITRY_NOT_FOUND: u64 = 1;
+
+    /// Default automation gas limit
+    const DEFAULT_AUTOMATION_GAS_LIMIT: u64 = 1000000;
+    /// Default upper gas duration in seconds
+    const DEFAULT_DURATION_UPPER_LIMIT: u64 = 2592000;
+    /// Registry resource creation seed
+    const REGISTRY_RESOURCE_SEED: vector<u8> = b"supra_framework::automation_registry";
+
+    /// It tracks entries both pending and completed, organized by unique indices.
+    struct AutomationRegistry has key, store {
+        /// The current unique index counter for registries. This value increments as new registries are added.
+        current_index: u64,
+        /// automation gas limit
+        automation_gas_limit: u64,
+        /// duration upper limit
+        duration_upper_limit: u64,
+        /// it's resource address which is use to deposit user automation fee
+        registry_fee_address: address,
+        /// resource account signature capability
+        registry_fee_address_signer_cap: SignerCapability,
+        /// A collection of automation task entries that are active state.
+        registries: table::Table<u64, AutomationTaskMetaData>,
+    }
+
+    #[event]
+    /// `AutomationTaskMetaData` represents a single automation task item, containing metadata.
+    struct AutomationTaskMetaData has copy, store, drop {
+        /// Automation task request id which is unique
+        id: u64,
+        /// The address of the registry owner.
+        owner: address,
+        /// The function signature associated with the registry entry.
+        payload_tx: vector<u8>,
+        /// Expiry of the registry entry, represented in a timestamp.
+        expiry_time: u64,
+        /// The transaction hash of the request transaction.
+        tx_hash: vector<u8>,
+        /// Max gas fee of automation task
+        max_gas: u64,
+        /// maximum gas price cap for the task
+        gas_price_cap: u64,
+        /// registration epoch number
+        registration_epoch: u64,
+        /// registration epoch time
+        registration_time: u64,
+        /// A boolean is_active indicating whether the registry entry start processing or not
+        is_active: bool
+    }
+
+    // todo : this function should call during initialzation, but since we already done genesis in that case who can access the function
+    fun initialize(supra_framework: &signer) {
+        system_addresses::assert_supra_framework(supra_framework);
+
+        let (registry_fee_resource_signer, registry_fee_address_signer_cap) = account::create_resource_account(
+            supra_framework,
+            REGISTRY_RESOURCE_SEED
+        );
+
+        move_to(supra_framework, AutomationRegistry {
+            current_index: 0,
+            automation_gas_limit: DEFAULT_AUTOMATION_GAS_LIMIT,
```

Comment:
I think there should be separate type for config holding these values, and it should be read during initialization.